src/tree_panel.py

```diff
@@ -1064,11 +1066,103 @@ def create_group(self):
             self.trvResources.setCurrentIndex
         )
 
+    def get_proxy_settings(self):
+        name_of_conn = NgwPluginSettings.get_selected_ngw_connection_name()
+        conn_sett = NgwPluginSettings.get_ngw_connection(name_of_conn)
+        proxies = None
+        auth = None
+        if conn_sett.proxy_enable:
+            proxies = {
+                'https': f'{conn_sett.proxy_host}:{conn_sett.proxy_port}',
+            }
+            auth = (f'{conn_sett.proxy_user}', f'{conn_sett.proxy_password}')
+        return {
+            'proxies': proxies,
+            'auth': auth
+        }
+
+    def get_connection_settings(self):
+        name_of_conn = NgwPluginSettings.get_selected_ngw_connection_name()
+        return NgwPluginSettings.get_ngw_connection(name_of_conn)
+
+    def need_to_check_service_plan_limits(self):
```

Comment:
Literally, it's `def user_has_a_basic_subscription` and it might be a cached property